src/main/java/net/sf/jabref/logic/cleanup/ISSNCleanup.java

```diff
@@ -0,0 +1,33 @@
+package net.sf.jabref.logic.cleanup;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.sf.jabref.model.FieldChange;
+import net.sf.jabref.model.entry.BibEntry;
+
+
+public class ISSNCleanup implements CleanupJob {
+
+    private static final Pattern ISSN_PATTERN_NODASH = Pattern.compile("^(\\d{4})(\\d{3}[\\dxX])$");
+    @Override
+    public List<FieldChange> cleanup(BibEntry entry) {
+        Optional<String> issn = entry.getFieldOptional("issn");
+        if (!issn.isPresent()) {
+            return Collections.emptyList();
+        }
+
+        Matcher matcher = ISSN_PATTERN_NODASH.matcher(issn.get());
```

Comment:
So your problem is that for this PR you want to make sense of strings which look like ISSNs but are not quite in the standard format, right? What about creating `static Optional<ISSN> fromString(text)` which returns an ISSN object if JabRef could make sense of the input. Then the ISSN class has a constructor accepting the different parts of the ISSN (first part, second part and checkdigit), `isValid` returns true if the checksum is correct and toString formats it in the standard way using a separating dash. Would this be a solution?